src/TypeVisitor.ts

```diff
@@ -563,4 +566,34 @@ export default class TypeVisitor {
     this.symbolTable.overwriteUpOne();
     this.symbolTable = this.symbolTable.getParentScope() as SymbolTable;
   }
+
+  private visitIfStatement(node: t.IfStatement) {
+    let testType = this.visitExpression(node.test);
+    if (!this.containsBoolean(testType)) {
```

Comment:
I don't think this check is needed - JS allows you to check truthiness of arbitrary values, e.g.

```
> if (2) { console.log("truthy") } else { console.log("falsey") }
truthy
undefined
```